hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java

```diff
@@ -337,6 +338,12 @@ public void deleteProjectIndices(Project project)
         LOG.log(Level.SEVERE, "Could not delete project index:{0}", index);
       }
     }
+  
+    // delete file provenance index for the project
+    String provenanceIndex = Provenance.getProjectIndex(project);
```

Comment:
@o-alex is that safe to do here, or there is a chance the index deletion will take too long and we'll get timeouts?

Reply:
So I think:
1. This will not be an issue for min provenance, might be an issue for full provenance
2. We should do a async index delete. This will not be an issue even in the case of delete/create/delete/create type of workflows, since I am not using project name, but project inode id in the index name. So there should be enough time to release the data until that inode id is to be reused.

Reply:
@o-alex: I guess that if ePipe pushed more documents after deleting the project the provenance cleaner will take care of deleting it again.